app/models/user.rb

```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_one :buyer
+  has_one :seller
```

Comment:
acho que um user não tem que ser obrigatoriamente um buyer e seller ao mesmo tempo, e pode mesmo nem ser nenhum, por isso estas relações deveriam ser `optional: true`

Reply:
Não encontrei essa config nas [opções](https://guides.rubyonrails.org/association_basics.html#options-for-has-one) para `has_one`. Um `user` sem `seller` ou `buyer` continua a ser um objeto válido. Pressuponho que seja opcional por defeito. Já um `buyer` ou `seller` tem de ter um `user`, é required por defeito.

Reply:
sim, tens razão, já é optional por defeito :) All good então